docs/schedule_scripting_language.md

```diff
@@ -0,0 +1,901 @@
+
+```c
+// Checks if the current time is between 18:00 ~ 6:00
+if_time_range (18, 0, 6, 0) {
+    return_time (18, 0, 6, 0, 2)
+} else {
+    return_none
+}
+```
+
+##### `if_time_range` non-generics
+
+`if_time_range_s` and `if_time_range_l`
+
+#### `if_misc`
+
+Checks if the passed miscellaneous argument is true.
+
+##### `if_misc` arguments
+
+- Argument 0: A value of the `ScheduleCheckMisc` enum.
+
+##### `if_misc` example
+
+```c
+if_misc (SCHEDULE_CHECK_MISC_MASK_ROMANI) {
+    return_s (33)
+} else {
+    return_s (34)
+}
+```
+
+##### `if_misc` non-generics
+
+`if_misc_s`. Note there's no long version of this command. It is advised to
+minimize the amount of commands used on a `ìf_misc` body and `else`'s body.
+
+#### `if_scene`
+
+Checks if the current scene matches the one passed as argument.
+
+##### `if_scene` arguments
+
+- Argument 0: A value of the `SceneId` enum.
+
+##### `if_scene` example
+
+```c
+if_scene (SCENE_SECOM) {
+    return_s (7)
+}
+```
+
+##### `if_scene` non-generics
+
+`if_scene_s` and `if_scene_l`
+
+#### `if_day`
+
+Checks if the current day matches the passed argument.
+
+##### `if_day` arguments
+
+- Argument 0: The day to check.
+
+##### `if_day` example
+
+```c
+if_day (3) {
+    return_l (1)
+}
+```
+
+##### `if_day` non-generics
+
+`if_day_s` and `if_day_l`
+
+#### `if_before_time`
+
+Checks if the current time is before the time passed as argument.
+
+##### `if_before_time` arguments
+
+- Argument 0: The hour component of the time.
+- Argument 1: The minute component of the time.
+
+##### `if_before_time` example
+
+```c
+if_before_time (8, 0) {
+    return_s (19)
+} else {
+    return_none
+}
+```
+
+##### `if_before_time` non-generics
+
+`if_before_time_s` and `if_before_time_l`
+
+##### `if_before_time` notes
+
+This command performs the opposite check of `if_since_time`.
+
+#### `if_since_time`
+
+Checks if the current time is after or equal to the time passed as argument.
+
+##### `if_since_time` arguments
+
+- Argument 0: The hour component of the time.
+- Argument 1: The minute component of the time.
+
+##### `if_since_time` example
+
+```c
+if_since_time (13, 0) {
+    return_s (9)
+} else {
+    return_none
+}
+```
+
+##### `if_since_time` non-generics
+
+`if_since_time_s` and `if_since_time_l`
+
+##### `if_since_time` notes
+
+This command performs the contrary check to `if_before_time`.
+
+### Unconditional branches
+
+Unconditional branches are basically the equivalent of C's `goto`s. They require
+a [label](#labels) to know where to branch to.
+
+#### `branch`
+
+Unconditionally transfer control to the passed label.
+
+##### `branch` arguments
+
+- Argument 0: The label to branch to.
+
+##### `branch` example
+
+```c
+if_day (3) {
+    if_since_time (10, 0) {
+        label_0x8:
+        return_none
+    } else {
+        branch (label_0xF)
+    }
+} else {
+    if_time_range (10, 0, 20, 0) {
+        branch (label_0x8)
+    } else {
+        label_0xF:
+        return_s (21)
+    }
+}
+```
```

Comment:
... the game actually does this in schedule format? its not just a bad match in the decompiled code, or is this just an example?

Reply:
This is an example taken verbatim from `src/overlays/actors/ovl_En_Door/D_808676F0.schedule`.

I converted all commands almost literally from their macro versions to this new language, and this script literally does this. You can see it yourself by following the control flow on the EnDoor actor:
https://github.com/zeldaret/mm/blob/10a14feb2c9a484597fce6e29df0ab5f7fa62848/src/overlays/actors/ovl_En_Door/z_en_door.c#L179-L185

As for how this script ended up having a control flow so weird, idk. Maybe they upgraded their tooling to have optimization passes (a lot of schedules don't seem to be optimized at all) or maybe the EnDoor dev was very passionate about optimizing door's schedules